src/Bicep.Core/SemanticModel/NameBindingVisitor.cs

```diff
@@ -121,19 +146,44 @@ private Symbol LookupSymbolByName(string name, TextSpan span)
             // symbol does not exist in the local namespace
             // try it in the imported namespaces
 
-            // match in one of the namespaces
-            var foundSymbols = this.namespaces
-                .Select(ns => ns.TryGetFunctionSymbol(name))
-                .Where(symbol => symbol != null)
-                .ToList();
+            // check if name is a namespace symbol first
+            var namespaceSymbol = this.namespaces
```

Comment:
>namespaces [](start = 39, length = 10)

If @namespace is "az" and name is "az", wouldn't this match the namespace also? That doesn't seem right.

Reply:
added an invalidExpression:

var invalidInstanceFunctionCall = az.az()

If we have something like:

From instance function call I think it is covered, there are two cases missing:

var someFoo = az // namespace assignment

and 

var someBar = az.az //property access